### PR TITLE
Render warning message as html

### PR DESCRIPTION
**Pull request by JPinkney**
Signed-off-by: Josh Pinkney <joshpinkney@gmail.com>

### What does this PR do?
This PR makes it so that anything in the warning message is rendered as html

~~CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23018~~ ended up not being needed

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18769

### CODE REVIEW

**Review by akurinnoy — APPROVED** [review]

**Review by olexii4 — APPROVED** [review]

File: src/components/BannerAlert/WarningBanner/index.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ class WarningBannerAlert extends React.PureComponent<Props, State> {
 
     return (
       <Banner className="pf-u-text-align-center" variant="warning">
-        {warningMessage}
+        <div dangerouslySetInnerHTML={{ __html: warningMessage }}></div>
```
Comment (sleshchenko) [inline src/components/BannerAlert/WarningBanner/index.tsx:32]: I'm OK with that as simplest option to go, assuming that the cluster admin is responsible for it's content. Then this is not more dangerous than have an ability to modify css, scripts on the packaging, or directly when you do kubectl exec )

**Review by sleshchenko — APPROVED** [review]

File: src/components/BannerAlert/WarningBanner/index.tsx (hunk 1 above)
Comment (sleshchenko) [reply src/components/BannerAlert/WarningBanner/index.tsx:32]: Well, later we will introduce an easy option to modify it for Che, CRW on CustomResource instance, but again, everybody who is able edit CheCluster CR - could do anything with custom dockerimage.

Comment (akurinnoy) [reply src/components/BannerAlert/WarningBanner/index.tsx:32]: We could sanitize the message from all tags except those we want to permit, using library like this one: https://www.npmjs.com/package/sanitize-html-react

**Comment by sleshchenko** [comment]
Maybe it makes sense to describe such an ability in https://github.com/eclipse/che-dashboard#configurability

**Review by ibuziuk — APPROVED** [review]
LGTM, indeed having a PR for docs would be great before merging this PR

**Comment by JPinkney** [comment]
Just waiting for the CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23018 then we can merge. `<a>` tags are allowed and `target` and `href` attributes are allowed, is there other tags/attributes needed?

**Comment by Ohrimenko1988** [comment]
[dashboard-next-ci-test]

**Comment by sleshchenko** [comment]
After https://github.com/eclipse/che-dashboard/pull/154 CQ is not needed anymore >
```diff
+| [`sanitize-html@2.3.2`](https://github.com/apostrophecms/sanitize-html.git) | MIT | clearlydefined |
```
Please regenerate deps files. after you pulled my changes.

File: package.json [hunk 2]
```diff
@@ -71,6 +71,7 @@
     "redux-thunk": "^2.3.0",
     "reflect-metadata": "^0.1.13",
     "reselect": "^4.0.0",
+    "sanitize-html": "^2.3.2",
```
Comment (sleshchenko) [inline package.json:74]: I see that we have a lot of ^ but it seems to be a mistake. We must update our dependencies manually, after making sure that's OK with CQs.

**Pull request merged**